lib/bpk-component-calendar/src/common-types-test.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @flow
+
+import {
+  assertString,
+  makeSingleSelection,
+  makeRangeSelection,
+  makeMultipleSelection,
+} from './common-types';
+
+describe('common-types', () => {
+  describe('assertStrings', () => {
+    it('throws when a key is missing', () => {
+      expect(() => {
+        assertString({ a: 'hello' }, 'b', 'myTest');
+      }).toThrow();
+    });
+
+    it('throws when a the value for a key is empty', () => {
+      expect(() => {
+        assertString({ a: 'hello', b: '' }, 'b', 'myTest');
+      }).toThrow();
+    });
+
+    it('does not throw when the key is present with a non-empty value', () => {
+      expect(() => {
+        assertString({ a: 'hello', b: 'test' }, 'b', 'myTest');
+      }).not.toThrow();
+    });
+  });
+
+  describe('makeSingleSelection', () => {
+    it('should return an appropriate object', () => {
+      expect(makeSingleSelection({ selectHint: 'Test' })).toEqual({
+        type: 'single',
+        selectHint: 'Test',
+      });
+    });
+
+    it('should throw when `selectHint` is missing', () => {
+      expect(() => {
+        // $FlowFixMe
```

Comment:
Is this due to the TODOs with the types and finding a better way to do it? Otherwise could you add a comment to say why this has been added here for future reference?

Reply:
No, it's becaues I am intentionally violating the flow types to cause the throw I want to test. I'll add a comment